Dockerfile

```
@@ -1,212 +1,58 @@
+#------------------------------------------------------------------------------
+RUN gem install bundler:2.0.2
 
-# Install alpine equivalent for "build-essential"
+#------------------------------------------------------------------------------
 #
-# and other dependencies required by dev.to various
-# ruby dependencies (postgresql-dev, tzdata)
-RUN apk add --no-cache alpine-sdk postgresql-dev tzdata
-
-# Im installing bash, as im a bash addict (not that great with sh)
-RUN apk add bash
-
-# Let's setup the rails directory
-# (@TODO - consider a production version?)
-WORKDIR /usr/src/app
-ENV RAILS_ENV development
-
-#####################################################
+# Install Yarn
 #
-# Let's prepare the dev.to source code files
-# WITHOUT docker related files
-#
-# This allow us to modify the docker
-# entrypoint / run file without recompiling
-# the entire application
-# (especially when creating this build script =| )
-#
-# (@TODO - improve and review ignore to blacklist unneeded items)
-#
-#####################################################
+#------------------------------------------------------------------------------
+ENV PATH=/root/.yarn/bin:$PATH
+RUN apk add --no-cache --repository=http://dl-cdn.alpinelinux.org/alpine/edge/community \
+        yarn
 
+#------------------------------------------------------------------------------
 #
-# Prepare the source code and remove any unneeded files
+# Define working directory
 #
-FROM alpine-ruby-node AS source-code-repo
-
-# The workdir
+#------------------------------------------------------------------------------
 WORKDIR /usr/src/app
-# Copy source code
-COPY ./ /usr/src/app/
-# remove docker related files
-RUN rm Dockerfile && rm docker-*
 
+#------------------------------------------------------------------------------
 #
-# Does the source code build
+# Copy Gemfile and run bundle install
 #
-FROM alpine-ruby-node AS source-code-build
-
-# Copy over files
-COPY --from=source-code-repo /usr/src/app/ /usr/src/app/
-
-# Run the various installer
-RUN gem install bundler
+#------------------------------------------------------------------------------
+COPY Gemfile /usr/src/app/Gemfile
+COPY Gemfile.lock /usr/src/app/Gemfile.lock
 RUN bundle install --jobs 20 --retry 5
-RUN yarn install && yarn check --integrity
 
-#####################################################
+#------------------------------------------------------------------------------
 #
-# Let's build the DEMO dev.to image
+# Now copy the app directory
 #
-#####################################################
-FROM alpine-ruby-node
-
-# Copy over the application code (without docker related files)
-COPY --from=source-code-build /usr/src/app/ /usr/src/app/
-
-# Copy over docker related files
-COPY Dockerfile [(docker-)]* /usr/src/app/
+#------------------------------------------------------------------------------
+COPY . /usr/src/app
```

Lets not add the whole directory here just to install the js dependencies this will, should we edit a file that is neither `package.json` or `yarn.lock` and build the container anew docker would invalidate the cache here which is unnecessary.  I'd say we add the rest of the files as late as possible. 
```suggestion
COPY ./package.json ./yarn.lock ./
```